### ISSUE TITLE
`useBroadcastEvent` - generic event payload

### ISSUE THREAD

**Issue by Pondorasti** [liveblocks/liveblocks#304]
**Is your feature request related to a problem? Please describe.**

`useBroadcastEvent` has poor typescript support and doesn't enforce what kind of payload it sends. Currently, it's type definition for `event` is marked as `any` 💀 :
```typescript
declare const useBroadcastEvent: () => (event: any, options?: _liveblocks_client_shared.B | undefined) => void;
```

**Describe the solution you'd like**

Add generics support similar to the implementation of `useEventListener`. Bellow is an example of how the generated `index.d.ts` would look:

```typescript
declare const useBroadcastEvent: <TEvent extends _liveblocks_client_shared.J>() => (event: TEvent, options?: _liveblocks_client_shared.B | undefined) => void;

// ------------

declare const useEventListener: <TEvent extends _liveblocks_client_shared.J>(callback: ({ connectionId, event, }: {
    connectionId: number;
    event: TEvent;
}) => void) => void;
```

**Comment by nvie** [comment]
Hi @Pondorasti, I'm glad you asked! We're currently in the process of planning our 0.17 release, which will bring a lot of extra TypeScript coverage/support, and `useBroadcastEvent` will also be included in those updates! 🎉

**Comment by Pondorasti** [comment]
Awesome, looking forward to the release! 🚢

**Comment by nvie** [comment]
@Pondorasti We have some good news! We're almost done with the 0.17 release, and we'll launch it publicly next week. If you're interested and have the time for it, you could help us test it and give feedback by trying out the beta release.

All the rationale and upgrade instructions can be found in our guide: [Upgrading to 0.17](https://preview.liveblocks.io/docs/guides/upgrading/0.17).

Any and all feedback would be much appreciated! 🙏  Thanks for your patience with this release ❤️

**Comment by nvie** [comment]
Closing this, as it's been fixed with the 0.17 release. Let me know if you run into any further trouble with this, @Pondorasti 🙏 !
